pyccel/codegen/printing/ccode.py

```diff
@@ -1185,6 +1199,11 @@ def formatted_args_to_printf(args_format, args, end):
                                                 file=expr.file)],
                                  unravelled = True)
                 code += self._print(body)
+            elif isinstance(f.class_type, StringType):
```

Comment:
Maybe simpler to just do:

```suggestion
            elif isinstance(f, LiteralString):
```
?